### ISSUE TITLE
why is there a need to seperate detection loss and id loss in such a complex way?

### ISSUE THREAD

**Issue by faruknane** [ifzhang/FairMOT#213]
Hi @ifzhang and the people in this community,
I try to add one more head to fairmot. I have successfully added it to the model and prepare the data loader according to my new dataset. However, in loss function there is something below that confuses me. 

![image](https://user-images.githubusercontent.com/37745467/90411897-15f06580-e0b5-11ea-92a6-0fb1448994e3.png)

I can't understand the reason behind separating the loss into two parts (detection and id) in such a complex way. You could have just multiplied id_loss and det_loss with "constant" coefficient factors to adjust the training to care more about one loss and less about the other one. You made s_det and s_id trainable parameters as coefficients for det_loss and id_loss. Why are those trainable? What you did is also similar to regularization, adding parameters to loss. Is it to prevent s_det and s_id from converging to zero?

**Comment by ifzhang** [comment]
I follow the multi-task learning uncertainly loss paper that can balance multiple loss:
https://arxiv.org/abs/1705.07115

**Comment by faruknane** [comment]
@ifzhang Thank you for your quick reply.
